compiler/testData/diagnostics/tests/controlStructures/ifToAnyDiscriminatingUsages.fir.kt

```diff
@@ -27,7 +27,7 @@ fun testResultOfAnonFun2() =
 
 fun testReturnFromAnonFun() =
         run(fun () {
-            return <!ARGUMENT_TYPE_MISMATCH, RETURN_TYPE_MISMATCH!>if (true) 42 else println()<!>
+            return if (true) 42 else println()
```

Comment:
Here the return type of `if-else` is constrained as `ILT:42 <: K <: Unit`, which results in `K = Unit` with a constraint error that `ILT:42 </: Unit`. However, thanks to `transformReturnExpression` using the `ResolutionMode.WithExpectedType(expectedTypeRef, expectedTypeMismatchIsReportedInChecker = true)` mode, the constraint error is discarded with the assumption that `FirFunctionReturnTypeMismatchChecker` will return a different error instead. It will not - the return type of the `if` is correct.

IMO `FirFunctionReturnTypeMismatchChecker` needs to be redesigned. Too many hacks in there and around it.

Reply:
Added another hack to return this particular error...